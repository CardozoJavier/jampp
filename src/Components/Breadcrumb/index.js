export { default as Breadcrumb } from './Breadcrumb';
export { default as Step } from './Step';