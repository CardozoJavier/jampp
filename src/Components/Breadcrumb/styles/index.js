export {
  BreadcrumbContainer,
  StepContainer,
  IconContainer,
  IconCircle,
  StepText,
  Step,
} from './Breadcrumb.styles';
