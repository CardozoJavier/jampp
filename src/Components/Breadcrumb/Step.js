import React from 'react';
import PropTypes from 'prop-types';
import { ArrowIcon, CheckIcon } from '../UI/Icons';
import { StepContainer, IconContainer, IconCircle, StepText } from './styles';

/**
 * Step component should be called with
 * @param {String} text - (Required) It's the name step.
 * @param {Boolean} isDone - (Optional) Define if the step is done for styling breadcrumb. 
 * @param {Boolean} isCurrent - (Optional) Define if is the current step. 
 * @param {Boolean} isLast - (Optional) Define if is the last step and not display ArrowIcon.
 * @param {String} borderColor - (Optional) Define the border color of check icon regarding step is done/current/next.
 */
const Step = ({ text, isDone = false, isCurrent = false, isLast = false, borderColor = '' }) => (
  <StepContainer>
    <IconContainer>
      <CheckIcon />
      <IconCircle borderColor={borderColor} isDone={isDone} />
    </IconContainer>
    <StepText isCurrent={isCurrent}>{text}</StepText>
    {!isLast &&
      <ArrowIcon />
    }
  </StepContainer>
);

Step.propTypes = {
  text: PropTypes.string.isRequired,
  isDone: PropTypes.bool.isRequired,
  isCurrent: PropTypes.bool.isRequired,
  isLast: PropTypes.bool.isRequired,
  borderColor: PropTypes.string.isRequired,
};

export default Step;
