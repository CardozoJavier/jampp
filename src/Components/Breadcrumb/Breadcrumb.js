import React from 'react';
import Step from './Step';
import { BreadcrumbContainer } from './styles';
import { palette } from '../styles';
const { green, gray, action } = palette;

const setBorderColor = (index, currentStep) => {
  let color;
  if (currentStep > index) {
    color = green.g1;
  } else if (currentStep === index) {
    color = action;
  } else {
    color = gray.g07;
  }

  return color;
}

const Breadcrumb = ({ children, currentStep }) => (
  <BreadcrumbContainer>
    {children.map((step, index) => (
      <Step
        isLast={index === (children.length-1)}
        isCurrent={currentStep === index}
        isDone={currentStep > index}
        isNext={currentStep < index}
        text={step.props.text}
        borderColor={setBorderColor(index, currentStep)}
      />
    ))}
  </BreadcrumbContainer>
);

export default Breadcrumb;
