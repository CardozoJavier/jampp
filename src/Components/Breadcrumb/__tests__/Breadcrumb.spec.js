/* eslint-disable no-undef */
import React from 'react';
import { render, screen } from '@testing-library/react';
import Breadcrumb from '../Breadcrumb';

describe('<Breadcrumb /> component', () => {
  const children = [
    <div key="key1" text="text 1" className="testing-length" />,
    <div key="key2" text="text 2" className="testing-length" />,
    <div key="key3" text="text 3" className="testing-length" />,
    <div key="key4" text="text 4" className="testing-length" />,
  ];
  test('should render children', () => {
    render(<Breadcrumb>{children}</Breadcrumb>);
    expect(screen.getAllByText(/text/)).toHaveLength(children.length);
  });

  test('should match snapshot', () => {
    const { container } = render(<Breadcrumb>{children}</Breadcrumb>);
    expect(container.firstChild).toMatchSnapshot();
  });
});
