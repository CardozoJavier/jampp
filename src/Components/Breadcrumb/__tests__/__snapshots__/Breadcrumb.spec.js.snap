// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Breadcrumb /> component should match snapshot 1`] = `
<div
  class="sc-jWBwVP jZcPPp"
>
  <div
    class="sc-brqgnP jyxKUE"
  >
    <div
      class="sc-cMljjf ldYMkv"
    >
      <div
        class="sc-htpNat sc-jTzLTM cBwkxP"
        disabled=""
      >
        <svg
          fill="none"
          height="9"
          viewBox="0 0 14 10"
          width="11"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 5L4.68 8.5L12.5657 1"
            stroke-width="2"
          />
        </svg>
      </div>
      <div
        class="sc-jAaTju gRHrAW"
      />
    </div>
    <p
      class="sc-jDwBTQ iiTNQM"
    >
      text 1
    </p>
    <div
      class="sc-htpNat sc-fjdhpX jtImPi"
    >
      <svg
        fill="none"
        height="8"
        viewBox="0 0 32 8"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.3536 4.35355C31.5488 4.15829 31.5488 3.84171 31.3536 3.64645L28.1716 0.464466C27.9763 0.269204 27.6597 0.269204 27.4645 0.464466C27.2692 0.659728 27.2692 0.976311 27.4645 1.17157L30.2929 4L27.4645 6.82843C27.2692 7.02369 27.2692 7.34027 27.4645 7.53553C27.6597 7.7308 27.9763 7.7308 28.1716 7.53553L31.3536 4.35355ZM0 4.5H31V3.5H0V4.5Z"
          fill="#9A9EA9"
        />
      </svg>
    </div>
  </div>
  <div
    class="sc-brqgnP jyxKUE"
  >
    <div
      class="sc-cMljjf ldYMkv"
    >
      <div
        class="sc-htpNat sc-jTzLTM cBwkxP"
        disabled=""
      >
        <svg
          fill="none"
          height="9"
          viewBox="0 0 14 10"
          width="11"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 5L4.68 8.5L12.5657 1"
            stroke-width="2"
          />
        </svg>
      </div>
      <div
        class="sc-jAaTju hwtDyn"
      />
    </div>
    <p
      class="sc-jDwBTQ ktgbGO"
    >
      text 2
    </p>
    <div
      class="sc-htpNat sc-fjdhpX jtImPi"
    >
      <svg
        fill="none"
        height="8"
        viewBox="0 0 32 8"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.3536 4.35355C31.5488 4.15829 31.5488 3.84171 31.3536 3.64645L28.1716 0.464466C27.9763 0.269204 27.6597 0.269204 27.4645 0.464466C27.2692 0.659728 27.2692 0.976311 27.4645 1.17157L30.2929 4L27.4645 6.82843C27.2692 7.02369 27.2692 7.34027 27.4645 7.53553C27.6597 7.7308 27.9763 7.7308 28.1716 7.53553L31.3536 4.35355ZM0 4.5H31V3.5H0V4.5Z"
          fill="#9A9EA9"
        />
      </svg>
    </div>
  </div>
  <div
    class="sc-brqgnP jyxKUE"
  >
    <div
      class="sc-cMljjf ldYMkv"
    >
      <div
        class="sc-htpNat sc-jTzLTM cBwkxP"
        disabled=""
      >
        <svg
          fill="none"
          height="9"
          viewBox="0 0 14 10"
          width="11"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 5L4.68 8.5L12.5657 1"
            stroke-width="2"
          />
        </svg>
      </div>
      <div
        class="sc-jAaTju hwtDyn"
      />
    </div>
    <p
      class="sc-jDwBTQ ktgbGO"
    >
      text 3
    </p>
    <div
      class="sc-htpNat sc-fjdhpX jtImPi"
    >
      <svg
        fill="none"
        height="8"
        viewBox="0 0 32 8"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M31.3536 4.35355C31.5488 4.15829 31.5488 3.84171 31.3536 3.64645L28.1716 0.464466C27.9763 0.269204 27.6597 0.269204 27.4645 0.464466C27.2692 0.659728 27.2692 0.976311 27.4645 1.17157L30.2929 4L27.4645 6.82843C27.2692 7.02369 27.2692 7.34027 27.4645 7.53553C27.6597 7.7308 27.9763 7.7308 28.1716 7.53553L31.3536 4.35355ZM0 4.5H31V3.5H0V4.5Z"
          fill="#9A9EA9"
        />
      </svg>
    </div>
  </div>
  <div
    class="sc-brqgnP jyxKUE"
  >
    <div
      class="sc-cMljjf ldYMkv"
    >
      <div
        class="sc-htpNat sc-jTzLTM cBwkxP"
        disabled=""
      >
        <svg
          fill="none"
          height="9"
          viewBox="0 0 14 10"
          width="11"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 5L4.68 8.5L12.5657 1"
            stroke-width="2"
          />
        </svg>
      </div>
      <div
        class="sc-jAaTju hwtDyn"
      />
    </div>
    <p
      class="sc-jDwBTQ ktgbGO"
    >
      text 4
    </p>
  </div>
</div>
`;
